### PR TITLE
Replace jcabi-mysql-maven-plugin with H2 for testing

### PR DESCRIPTION
**Pull request by wcekan**
Replacing a deprecated `jcabi-mysql-maven-plugin` with H2 for database testing.

### CODE REVIEW

File: elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java [hunk 1]
```diff
@@ -1329,12 +1329,12 @@ public void patchExtBadValue() throws IOException {
...
-        Assert.assertNotNull(errors);
-        Assert.assertEquals(errors.size(), 1);
+        assertNotNull(errors);
+        assertEquals(errors.size(), 1);
 
         String error = errors.get(0).asText();
-        String expected = "TransactionException: Duplicate entry 'duplicate' for key 'name'";
-        Assert.assertTrue(error.equals(expected), "Error does not equal with '" + expected + "'");
+        String expected = "TransactionException: Unique index or primary key violation:";
+        assertTrue(error.startsWith(expected), "Error does not start with '" + expected + "' but found " + error);
```
Comment (DennisMcWherter) [inline elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java]: minor question: why the change here from `equals` to `startsWith`?

Comment (wcekan) [reply elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java]: The actual message is 
```
TransactionException: Unique index or primary key violation: "CONSTRAINT_INDEX_5 ON PUBLIC.CHILDENTITY(NAME) VALUES (CAST('duplicate' AS VARCHAR_IGNORECASE), 10)"
```

Comment (wcekan) [reply elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java]: Additional change from #739 because Hibernate 3 and 5 throw different error strings.
